dateparser/date.py

```diff
@@ -175,7 +175,7 @@ def _try_given_formats(self):
         if not self.date_formats:
             return
 
-        return parse_with_formats(self._get_translated_date_with_formatting(), self.date_formats)
+        return parse_with_formats(self._get_translated_date_with_formatting(), list(self.date_formats))
```

Comment:
It's not garunteed to be an iterable and in the event that it isn't it used
to to something foolish in the end as well as iterate over the string.

On Mon, Feb 9, 2015, 1:15 PM Eugene Amirov <notifications@github.com> wrote:

> In dateparser/date.py
> <https://github.com/scrapinghub/dateparser/pull/18#discussion_r24350439>:
>
> > @@ -175,7 +175,7 @@ def _try_given_formats(self):
> >          if not self.date_formats:
> >              return
> >
> > -        return parse_with_formats(self._get_translated_date_with_formatting(), self.date_formats)
> > +        return parse_with_formats(self._get_translated_date_with_formatting(), list(self.date_formats))
>
> Could you please remind me why do we need to construct list here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scrapinghub/dateparser/pull/18/files#r24350439>.
>